File: corporate-reporting/src/types/PartnerOrganisationUpdate.ts
import { IPartnerOrganisation } from "./PartnerOrganisation";
import { IRagOption } from "./RagOption";
import { IReportingProgressUpdate, ReportingProgressUpdate } from "./ReportingProgressUpdate";

export interface IPartnerOrganisationUpdate extends IReportingProgressUpdate {
    PartnerOrganisationID: number;
    OverallRagOptionID: number;
    FinanceRagOptionID: number;
    FinanceComment: string;
    PeopleRagOptionID: number;
    PeopleComment: string;
    MilestonesRagOptionID: number;
    MilestonesComment: string;
    KPIRagOptionID: number;
    KPIComment: string;
    ProgressUpdate: string;
    FutureActions: string;
    Escalations: string;
    PartnerOrganisation?: IPartnerOrganisation;
    OverallRagOption?: IRagOption;
    FinanceRagOption?: IRagOption;
    PeopleRagOption?: IRagOption;
    MilestonesRagOption?: IRagOption;
    MetricsRagOption?: IRagOption;
}

export class PartnerOrganisationUpdate extends ReportingProgressUpdate implements IPartnerOrganisationUpdate {
    public PartnerOrganisationID = null;
    public OverallRagOptionID = null;
    public FinanceRagOptionID = null;
    public FinanceComment = '';
    public PeopleRagOptionID = null;
    public PeopleComment = '';
    public MilestonesRagOptionID = null;
    public MilestonesComment = '';
    public KPIRagOptionID = null;
    public KPIComment = '';
    public ProgressUpdate = '';
    public FutureActions = '';
    public Escalations = '';

    public PartnerOrganisation = null;
    public UpdateUser = null;

    constructor(period: Date, partnerOrganisationId?: number, partnerOrganisation?: IPartnerOrganisation) {
        super(period);
        if (partnerOrganisationId) this.PartnerOrganisationID = partnerOrganisationId;
        if (partnerOrganisation) this.PartnerOrganisation = partnerOrganisation;
    }
}

File: ORB.Core/Models/PartnerOrganisationUpdate.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class PartnerOrganisationUpdate : EntityUpdate
    {
        public string Comment { get; set; }
        public int? RagOptionID { get; set; }
        public int? PartnerOrganisationID { get; set; }
        public int? OverallRagOptionID { get; set; }
        public int? FinanceRagOptionID { get; set; }
        public string FinanceComment { get; set; }
        public int? PeopleRagOptionID { get; set; }
        public string PeopleComment { get; set; }
        public int? MilestonesRagOptionID { get; set; }
        public string MilestonesComment { get; set; }
        public int? KPIRagOptionID { get; set; }
        public string KPIComment { get; set; }
        public string ProgressUpdate { get; set; }
        public string FutureActions { get; set; }
        public string Escalations { get; set; }
        public bool? ToBeClosed { get; set; }
        public DateTime? UpdatePeriod { get; set; }

        public virtual RagOption FinanceRagOption { get; set; }
        public virtual RagOption KPIRagOption { get; set; }
        public virtual RagOption MilestonesRagOption { get; set; }
        public virtual RagOption OverallRagOption { get; set; }
        public virtual PartnerOrganisation PartnerOrganisation { get; set; }
        public virtual RagOption PeopleRagOption { get; set; }
    }
}
